### PR TITLE
automatically clean-up old archives, upload max. one archive per day

### PR DESCRIPTION
**Pull request by IsaackMwenda**
copied from Imaqal-Covid19

### CODE REVIEW

File: src/lib/pipeline_configuration.py [hunk 1]
```diff
@@ -745,21 +749,13 @@ def validate(self):
...
 
-        validators.validate_string(self.memory_profile_upload_url_prefix, "memory_profile_upload_url_prefix")
-
-        if self.listening_group_csv_urls is not None:
-            validators.validate_list(self.listening_group_csv_urls, "listening_group_csv_urls")
-            for i, listening_group_csv_url in enumerate(self.listening_group_csv_urls):
-                validators.validate_string(listening_group_csv_url, f"{listening_group_csv_url}")
-
+        memory_profile_upload_url_prefix = f"{self.memory_profile_upload_bucket}{self.bucket_dir_path}"
+        validators.validate_string(memory_profile_upload_url_prefix, "memory_profile_upload_url_prefix")
```
Comment (as2388) [inline src/lib/pipeline_configuration.py]: Better to validate_url with the gs test, and to validate all of the new variables

File: upload_log_files.py [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+            if file_date not in files_for_days_that_upload_failed:
+                files_for_days_that_upload_failed[file_date] = []
+
+            files_for_days_that_upload_failed[file_date].append(file_path)
+
+    # Check for latest modified file path for each day that failed to upload
+    # Delete other files for that date
+    for file_date in files_for_days_that_upload_failed:
+        for file_path in files_for_days_that_upload_failed[file_date]:
+            if file_path == max(files_for_days_that_upload_failed[file_date], key=os.path.getmtime):
```
Comment (as2388) [inline upload_log_files.py]: This will fail if you delete a file in this list

**Review by as2388 — APPROVED** [review]

File: upload_log_files.py (hunk 2 above)
Comment (IsaackMwenda) [reply upload_log_files.py]: updated thanks.

File: src/lib/pipeline_configuration.py (hunk 1 above)
Comment (IsaackMwenda) [reply src/lib/pipeline_configuration.py]: updated thanks.

**Pull request merged**